File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/FileUpload.py
from __future__ import annotations
import io
from src.main.org.apache.commons.fileupload.FileItemFactory import *
from src.main.org.apache.commons.fileupload.FileUploadBase import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class FileUpload(FileUploadBase, metaclass=StaticFieldRedirector):
    def setFileItemFactory(self, factory: FileItemFactory) -> None:

        idMapPyToJ = self.javaObj.pyToJ()
        translatedArg0 = JavaHandler.valueToObject(factory, "FileItemFactory", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj.setFileItemFactory(translatedArg0)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            JavaHandler.mapping(translatedArg0, idMapJToPy, factory)
    def getFileItemFactory(self) -> FileItemFactory:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.getFileItemFactory(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def __init__(self, constructorId: int, fileItemFactory: FileItemFactory) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(constructorId, "int", idMapPyToJ)
        translatedArg1 = JavaHandler.valueToObject(fileItemFactory, "FileItemFactory", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj = FileUpload.javaClz(translatedArg0, translatedArg1)
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            JavaHandler.mapping(translatedArg0, idMapJToPy, constructorId)
            JavaHandler.mapping(translatedArg1, idMapJToPy, fileItemFactory)
    javaClz = java.type("org.apache.commons.fileupload.FileUpload")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))

File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/FileUpload.java
package org.apache.commons.fileupload;

import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public class FileUpload extends FileUploadBase {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/FileUpload.py", "FileUpload"));
  public FileItemFactory fileItemFactory;

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  @Override
  public void setFileItemFactory(FileItemFactory factory) {

    this.fileItemFactory = factory;
  }

  @Override
  public FileItemFactory getFileItemFactory() {

    return fileItemFactory;
  }

  public FileUpload(int constructorId, FileItemFactory fileItemFactory) {

    super();
    if (constructorId == 1) {
      this.fileItemFactory = fileItemFactory;
    }
  }

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {
    super.pyToJ(idMap);
    this.fileItemFactory =
        (FileItemFactory)
            IntegrationUtils.valueToObject(
                this.obj.getMember("_FileUpload__fileItemFactory"),
                "FileItemFactory",
                idMap,
                fileItemFactory);
    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {
    super.jToPy(idMap);
    this.obj.invokeMember("__setattr__", "javaObj", this);
    this.obj.invokeMember(
        "__setattr__",
        "_FileUpload__fileItemFactory",
        IntegrationUtils.mapToPython(
            fileItemFactory, idMap, this.obj.getMember("_FileUpload__fileItemFactory")));
    return idMap;
  }
}
